presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchingVisitor.java

```diff
@@ -38,43 +42,115 @@
     }
 
     @Override
+    public Boolean visitExchange(ExchangeNode node, PlanMatchingContext context)
+    {
+        checkState(node.getType() == ExchangeNode.Type.GATHER, "Only GATHER is supported");
+        List<List<Symbol>> allInputs = node.getInputs();
+        checkState(allInputs.size() == 1, "Multiple lists of inputs are not supported yet");
+
+        List<Symbol> inputs = allInputs.get(0);
+        List<Symbol> outputs = node.getOutputSymbols();
+
+        ImmutableMap.Builder<Symbol, Expression> assignments = ImmutableMap.builder();
```

Comment:
Please move building assignments to inside of https://github.com/Teradata/presto/pull/412/commits/bac3e33fed6b3c7a40120b0a0d06ca78f92f8ebd#diff-f0210b2ef21b3dc9287ffbed3a37199bR60 since it's not needed before or when `result == false`.